### PR TITLE
Refactor api page helper

### PR DESCRIPTION
**Pull request by violettine**
refactored helper and views of topic weather_api to lib file and controller. (Rails way)

### CODE REVIEW

File: app/helpers/curr_weather_helper.rb [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 
-  def cloud_view(curr_weather_data)
-    all_clouds_number = curr_weather_data['clouds']['all']
-    if all_clouds_number.zero?
-      @sky_view = 'sky is clear'
-    else
-      @sky_view = 'sky is not clear'
-    end
+  def kelvin_to_celsius(curr_data)
+    # converting kelv temo into celc
```
Comment (DemitryT) [inline app/helpers/curr_weather_helper.rb]: temp*

File: app/models/get_data_from_weather_api.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class GetDataFromWeatherApi
```
Comment (DemitryT) [inline app/models/get_data_from_weather_api.rb]: WeatherRequest or WeatherApiRequest would be a better name here I think

File: app/models/weather_response.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+class WeatherResponse
+  def initialize(params = {})
+    @curr_city = 'Berlin'
+    @forecast_city = 'Berlin'
+    if params[:curr]
+      @curr_city = params[:curr][:city]
+    elsif params[:forecast]
+      @forecast_city = params[:forecast][:city]
+    end
+    @existent_city = true
```
Comment (DemitryT) [inline app/models/weather_response.rb]: I think you might mean `existing_city`

File: app/models/weather_response.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  def same_forecast_city?(cache_name)
+    forecast = Rails.cache.fetch(cache_name)
+    (forecast['city']['name'].eql?(@forecast_city.capitalize))
+  end
+
+  def get_weather_api_data
+    forecast_period = '12'
+    data = GetDataFromWeatherApi.new(@curr_city, @forecast_city, forecast_period)
+
+    if (data.current['cod'] == '404' || data.forecast['cod'] == '404')
```
Comment (DemitryT) [inline app/models/weather_response.rb:41]: They spell it `cod` instead of `code`?

File: spec/controllers/api_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    it 'renders the personal template' do
+      get :index
+      expect(response).to render_template('index')
+    end
+  end
+
+  describe 'POST #index' do
+   it 'responds successfully with an HTTP 200 status code' do
+      post :index
```
Comment (DemitryT) [inline spec/controllers/api_controller_spec.rb:19]: you're usually posting with some parameters though, right? Might be good to test that as well

File: app/models/weather_response.rb (hunk 4 above)
Comment (violettine) [inline app/models/weather_response.rb:41]: Strange but true :D

**Pull request merged**